src/encoded/static/components/item-pages/MicroscopeConfigurationView.js

```diff
@@ -0,0 +1,412 @@
+            'addFileLoading'        : false,
+            'modal'                 : null
+        };
+
+        this.microMetaToolRef = React.createRef();
+    }
+
+    componentDidMount(){
+        const onComplete = () => {
+            this.setState({ mounted: true });
+        };
+
+        if (!MicroscopyMetadataTool) {
+            setTimeout(()=>{
+                // Load in HiGlass libraries as separate JS file due to large size.
+                // @see https://webpack.js.org/api/module-methods/#requireensure
+                require.ensure(['4dn-microscopy-metadata-tool'], (require) => {
+                    MicroscopyMetadataTool = require('4dn-microscopy-metadata-tool').default;
+                    onComplete();
+                }, "microscopy-metadata-bundle");
+            });
+        } else {
+            onComplete();
+        }
+    }
+
+    getMicroscopyMetadataToolComponent(){
+        return (this.microMetaToolRef && this.microMetaToolRef.current && this.microMetaToolRef.current) || null;
+    }
+
+    havePermissionToEdit(){
+        const { session, context : { actions = [] } } = this.props;
+        return !!(session && _.findWhere(actions, { 'name' : 'edit' }));
+    }
+
+    /**
+        * Update the current higlass viewconfig for the user, based on the current data.
+        * Note that this function is throttled in constructor() to prevent someone clicking it like, 100 times within 3 seconds.
+        * @returns {void}
+    */
+    handleSave(evt) {
+        const { href, context } = this.props;
+        const { modal } = this.state;
+        evt.preventDefault();
+
+        const mtc = this.getMicroscopyMetadataToolComponent();
+
+        if (!mtc || !mtc.api) {
+            throw new Error('Could not get current MicroscopyMetadataToolComponent api.');
+        }
+
+        if (!this.havePermissionToEdit()) {
+            // I guess would also get caught in ajax error callback.
+            throw new Error('No edit permissions.');
+        }
+
+        if (modal == null && context.status && typeof context.status === 'string' &&
+            (context.status === 'released' || context.status === 'released to project')) {
+            this.setState({
+                'modal': (
+                    <ConfirmModal handleConfirm={this.handleSave} handleCancel={this.handleModalCancel}
+                        confirmButtonText="Save" cancelButtonText="Cancel" modalTitle="Confirm Save">
+                        You are overwriting a Microscope Configuration Item that was previously shared with public. Are you sure?
+                        <br />Note that you can also clone this display and share the new copy.
+                    </ConfirmModal>
+                )
+            });
+            return;
+        }
+
+        mtc.api.saveMicroscope();
+    }
+
+    statusChangeButton(){
+        const { session, context } = this.props;
+        const { saveLoading, cloneLoading, releaseLoading } = this.state;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['cloneLoading' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4271272)